R/undesirable_function_linter.R

```diff
@@ -0,0 +1,58 @@
+default_undesirable_funs <- c(
+  "setwd"="use withr::with_dir()",
+  "sink"="use withr::with_sink()",
+  "par"="use withr::with_par()",
+  "options"="use withr::with_options()",
+  "Sys.setenv"="use withr::with_envvar()",
+  "Sys.setlocale"="use withr::with_locale()",
+  ".libPaths"="use withr::with_libpaths()",
+  "attach"="use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "detach"="use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "library"= "use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "require"= "use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "loadNamespace"="use `::` or requireNamespace()",
+  "source"=NA,
+  "sapply"="use vapply() or lapply()",
+  "mapply"="use Map()",
+  "ifelse"="use an if () {} else {} block",
```

Comment:
While it is true some (most?) of the time people use ifelse when it would be better to use `if()` there are cases where you do want to branch on a vectorized conditional, I think it is too aggressive to lint this.